File: 002,leetcode_py3/leetcode-20.py
class Solution:
	def isValid(self, s):
		pre_dict = {')':'(', ']':'[', '}':'{'}
		stack = []
		for i in s:
			if i in pre_dict.values():
				stack.append(i)
			elif not stack or pre_dict[i] != stack.pop():
				return False
		return not stack

File: 003,tencent-50/Java/leetcode-20.java
class Solution {
    public boolean isValid(String s) {
        
        Map<Character,Character> map = new HashMap<>();
        map.put('(',')');
        map.put('[',']');
        map.put('{','}');
        
        Stack<Character> stack = new Stack<Character>();  // 第二个的Character可以不写
        char[] charArray = s.toCharArray();
        
        for (Character c:charArray){
            if (map.keySet().contains(c)){
                stack.push(c);
            }else{
                if (!stack.isEmpty() && map.get(stack.peek()) == c){
                    stack.pop();
                }else{
                    return false;
                }
            }
        }
        
        return stack.isEmpty();
        
    }
}


